### PR TITLE
FreeBSD fixes

### PR DESCRIPTION
**Pull request by AMDmi3**

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -3,10 +3,10 @@ APPCLIENT=$(APPNAME)
 APPSERVER=$(APPNAME)_server
 
 #CXXFLAGS= -ggdb3
-CXXFLAGS= -O3 -fomit-frame-pointer -ffast-math
+CXXFLAGS?= -O3 -fomit-frame-pointer -ffast-math
```
Comment (qreeves) [inline src/Makefile]: What does this do?

File: src/enet/Makefile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CFLAGS=-O3 -fomit-frame-pointer
+CFLAGS?=-O3 -fomit-frame-pointer
```
Comment (qreeves) [inline src/enet/Makefile]: We don't maintain "eNet", you'll have to prod @lsalzman about this one.

File: src/Makefile (hunk 1 above)
Comment (AMDmi3) [inline src/Makefile]: CFLAGS/CXXFLAGS may be set by user or build environment and should not be overridden in that case. Default value is OK though. This change preserves default value, but respects user setting if there is one.

File: src/enet/Makefile (hunk 2 above)
Comment (AMDmi3) [inline src/enet/Makefile]: This Makefile is probably maintained by RE contributors, as it doesn't exist in enet repo.

Comment (qreeves) [inline src/enet/Makefile]: Actually, this file shouldn't be in the repo at all, it is generated by './configure', haha.

Comment (AMDmi3) [inline src/enet/Makefile]: There's no longer configure in the repo, no? The looks to be handwritten.

**Pull request closed without merging**